pages/thc/[processRef]/manager-review.tsx

```diff
@@ -0,0 +1,314 @@
+  const processDatabase = useDatabase(Storage.ProcessContext);
+
+  const tenants = useDataValue(
+    Storage.ExternalContext,
+    "residents",
+    processRef,
+    (values) => (processRef ? values[processRef]?.tenants : undefined)
+  );
+
+  const tenantsValue = tenants.result || [];
+
+  const tenantIdsPresentForCheck = useDataValue(
+    Storage.ProcessContext,
+    "tenantsPresent",
+    processRef,
+    (values) => (processRef ? values[processRef] : undefined)
+  );
+
+  const tenantIdsPresentForCheckValue = tenantIdsPresentForCheck.result || [];
+
+  const tenantsWithPresentStatus = tenantsValue.map((tenant) => ({
+    ...tenant,
+    present: tenantIdsPresentForCheckValue.includes(tenant.id),
+  }));
+
+  const address = useDataValue(
+    Storage.ExternalContext,
+    "residents",
+    processRef,
+    (values) => (processRef ? values[processRef]?.address : undefined)
+  );
+
+  const tenureType = useDataValue(
+    Storage.ExternalContext,
+    "tenancy",
+    processRef,
+    (values) => (processRef ? values[processRef]?.tenureType : undefined)
+  );
+
+  const tenancyStartDate = useDataValue(
+    Storage.ExternalContext,
+    "tenancy",
+    processRef,
+    (values) => (processRef ? values[processRef]?.startDate : undefined)
+  );
+
+  const officerFullName = useDataValue(
+    Storage.ExternalContext,
+    "officer",
+    processRef,
+    (values) => (processRef ? values[processRef]?.fullName : undefined)
+  );
+
+  const officerFullNameValue = officerFullName.result || "Loading...";
+
+  const submittedDate = useDataValue(
+    Storage.ProcessContext,
+    "submitted",
+    processRef,
+    (values) => (processRef ? values[processRef] : undefined)
+  );
+
+  const submittedDateValue = submittedDate.result
+    ? formatDate(new Date(submittedDate.result), "d MMMM yyyy")
+    : "Loading...";
+
+  const isUnannouncedVisit = useDataValue(
+    Storage.ProcessContext,
+    "isUnannouncedVisit",
+    processRef,
+    (values) => (processRef ? values[processRef] : undefined)
+  );
+
+  const getManagerSummaryText = (
+    yesValue: string,
+    noValue: string,
+    databaseValue: { value: string; notes: Notes }
+  ): string => {
+    if (databaseValue.value === "yes") {
+      return yesValue;
+    } else {
+      return databaseValue.notes.length > 0
+        ? `${noValue}. Note: ${databaseValue.notes[0].value}`
+        : `${noValue}`;
+    }
+  };
+
+  const [managerComment, setManagerComment] = useState("");
+
+  const visitType = isUnannouncedVisit.result
+    ? getManagerSummaryText(
+        "Unannounced",
+        "Announced",
+        isUnannouncedVisit.result
+      )
+    : "Loading...";
+
+  const isVisitInside = useDataValue(
+    Storage.ProcessContext,
+    "isVisitInside",
+    processRef,
+    (values) => (processRef ? values[processRef] : undefined)
+  );
+
+  const visitTookPlace = isVisitInside.result
+    ? getManagerSummaryText("Inside", "Outside", isVisitInside.result)
+    : "Loading...";
```

Comment:
Showing `Loading...` on `thing.result === undefined` is ok, but `thing.result` might also be `undefined` if there was an error, and that's hidden from the user by this. Nothing to do now, but something to be aware of.